script.js

```diff
@@ -1,8 +1,260 @@
+  const elem = document.createElement(tagname);
+  return elem;
+}
+
+//  Creates tag names with attributes and text
+function createTagWithAttribute(tagName, [attribute, value, text]) {
+  const tag = document.createElement(tagName);
+  tag.setAttribute(attribute, value);
+  tag.textContent = text;
+  return tag;
+}
+
+//  Creates tag names with only attributes
+function createTagWithAttributeOnly(tagName, [attribute, value]) {
+  const tag = document.createElement(tagName);
+  tag.setAttribute(attribute, value);
+  return tag;
+}
+
+function createCard(projects, index) {
+  //  create outer div for a card '.grid-item'
+  const card = createTag('div');
+  card.classList.add(...projects[index].cardClass);
+
+  // create inner div for a card
+  const cardImg = createTag('div');
+  cardImg.classList.add(...projects[index].cardImg);
+  card.appendChild(cardImg);
+
+  // create the project image tag
+  const imgTag = createTag('div');
+  imgTag.classList.add(...projects[index].imgDetails);
+  cardImg.appendChild(imgTag);
+
+  //  create the article
+  const article = createTag('div');
+  article.classList.add(...projects[index].articleClassList);
+  cardImg.appendChild(article);
+
+  //  create h2 title
+  const h3Title = createTag('div');
+  h3Title.classList.add(...projects[index].h2TitleClass);
+  article.appendChild(h3Title);
+
+  //  create title
+  const h3Content = createTag('h3');
+  h3Content.textContent = projects[index].name;
+  h3Title.appendChild(h3Content);
+
+  const canop = createTag('ul');
+  canop.classList.add('flex');
+  h3Title.appendChild(canop);
+
+  //  create options link;
+  const link1 = createTag('li');
+  link1.classList.add('canopy');
+  link1.textContent = 'CANOPY';
+  canop.appendChild(link1);
+
+  const link2 = createTag('li');
+  link2.textContent = 'Back End Dev';
+  canop.appendChild(link2);
+
+  const link3 = createTag('li');
+  link3.textContent = '2015';
+  canop.appendChild(link3);
+
+  //  create paragraph and append to article
+  const p = createTag('p');
+  p.textContent = projects[index].p;
+  article.appendChild(p);
+
+  //  create languge div ul
+  const langDiv = createTag('ul');
+  langDiv.classList.add(...projects[index].liFormat);
+  article.appendChild(langDiv);
+
+  //  create languge div links
+  const link4 = createTag('li');
+  link4.textContent = 'html';
+  langDiv.appendChild(link4);
+
+  const link5 = createTag('li');
+  link5.textContent = 'css';
+  langDiv.appendChild(link5);
+
+  const link6 = createTag('li');
+  link6.textContent = 'javascript';
+  langDiv.appendChild(link6);
+
+  //  create a button
+  const button = createTag('button');
+  button.type = 'submit';
+  button.name = index;
+  button.textContent = 'See Project';
+  article.appendChild(button);
+  cards.appendChild(card);
+}
+
+createCard(projects, 0);
+createCard(projects, 1);
+createCard(projects, 2);
+createCard(projects, 3);
+createCard(projects, 4);
+createCard(projects, 5);
+createCard(projects, 6);
```

Comment:
- [ ] For best javascript practices, please remove these lines of code and istead, write a loop for calling createCard() function for each element in your projects array.